src/sentry/integrations/services/integration/impl.py

```diff
@@ -584,3 +586,31 @@ def get_integration_identity_context(
             identity=identity,
             user=user,
         )
+
+    def refresh_github_access_token(
+        self, *, integration_id: int, organization_id: int
+    ) -> RpcIntegration | None:
+        try:
+            integration = Integration.objects.get(
+                id=integration_id,
+                provider__in=["github", "github_enterprise"],
+                status=ObjectStatus.ACTIVE,
+            )
+        except Integration.DoesNotExist:
+            return None
+
+        installation = integration.get_installation(organization_id=organization_id)
+        # get_installation doesn't actually check if the integration is
+        # associated with the organization, so this validates that it does,
+        # and caches the org_integration preemptively.
+        try:
+            installation.org_integration
+        except OrganizationIntegrationNotFound:
+            return None
+
+        installation.get_client().get_access_token(
+            token_minimum_validity_time=timedelta(minutes=10)
+        )
+        integration.refresh_from_db()
+
+        return serialize_integration(integration)
```

Comment:
Q, why not pass back the token? Would you need to make a new RpcModel to do that?